pkg/query/remote_engine.go

```diff
@@ -221,11 +221,25 @@ type remoteQuery struct {
 	interval   time.Duration
 	remoteAddr string
 
+	samplesStats *stats.QuerySamples
+
 	cancel context.CancelFunc
 }
 
 func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 	start := time.Now()
+	defer func() {
+		keys := []any{
+			"msg", "Executed remote query",
+			"query", r.plan.String(),
```

Comment:
I am a bit concerned that this will get executed even when the debug logger is disabled. But we can see if it ends up being a problem.

Reply:
Yeah, plus this was already there, just moved places and added sample information